### DISCUSSION TITLE
Is there a way to run post-upgrade commands for different managers?

### DISCUSSION THREAD

**Discussion by lyonlai** [renovatebot/renovate#30315] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

37.438.5

### Please tell us more about your question or problem

postUpgradeTasks at the moment is an object that contains a list of command to run for any managers.

Just wondering if it is possible to run the post-upgrade task specific to a manager. Say for example I have a repository with `bazel`, `hermit` and `npm`.  Ideally when only updating `hermit` packages, I don't need to run `bazel` or  `npm` lockfile updates. Same as doing `bazel`, no need to run `npm` lockfile update.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Have you tried using packageRules?

**Reply by lyonlai** [reply]
I didn't know that could go with packageRules. I will give it a try.

**Reply by ysangkok** [reply]
@lyonlai What did you end up doing?

**Reply by gzm0** [reply]
We use `postUpgradeTasks` nested inside `packageRules`. It works as you'd expect (watch out for mergeability though).
